src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,136 @@
+package org.jabref.logic.journals;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.net.URLDownload;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredatoryJournalLoader {
+    private static class PJSource {
+        URL url = null;
+        final Pattern elementPattern;
+
+        PJSource(String url, String regex) {
+            try {
+                this.url = new URL(url);
+            } catch (MalformedURLException ex) {
+                LOGGER.error("Malformed URL has occurred in PJSource", ex);
+            }
+            this.elementPattern = (regex != null) ? Pattern.compile(regex) : null;
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PredatoryJournalLoader.class);
+    private static final Pattern PATTERN_NAME = Pattern.compile("(?<=\">).*?(?=<)");
+    private static final Pattern PATTERN_URL = Pattern.compile("http.*?(?=\")");
+    private static final Pattern PATTERN_ABBR = Pattern.compile("(?<=\\()[^ ]*(?=\\))");
+    private static final List<String> LINK_ELEMENTS = new ArrayList<>();
+    private static final List<PJSource> PREDATORY_SOURCES = List.of(
+            new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/journals.csv",
+                    null),
+            new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/publishers.csv",
+                    null),
+            new PJSource("https://beallslist.net/",
+                    "<li>.*?</li>"),
+            new PJSource("https://beallslist.net/standalone-journals/",
+                    "<li>.*?</li>"),
+            new PJSource("https://beallslist.net/hijacked-journals/",
+                    "<tr>.*?</tr>")
+    );
+    private PredatoryJournalRepository repository = new PredatoryJournalRepository();
+
+    public static PredatoryJournalRepository loadRepository() {
+        PredatoryJournalLoader loader = new PredatoryJournalLoader();
+        // Initialize with built-in list
+        try (InputStream resourceAsStream = JournalAbbreviationRepository.class.getResourceAsStream("/journals/predatoryJournalList.mv")) {
+            if (resourceAsStream == null) {
+                LOGGER.warn("There is no predatoryJournalList.mv. We use a default predatory journal list");
+            } else {
+                Path tempDir = Files.createTempDirectory("jabref-journal");
```

Comment:
Instead of storing this in a temp directory, it's better to store this in the user's app data dir, so it's more permanent. See for example  for the log diretory of the backup directory how it's done
https://github.com/JabRef/jabref/blob/2a9e37c24bbf5bce6bf5e1547191afcffb3f6db9/src/main/java/org/jabref/gui/desktop/os/NativeDesktop.java#L88-L93

Reply:
That's a really good suggestion; I will switch to that.